config/routes.rb

```diff
@@ -1,6 +1,13 @@
 Rails.application.routes.draw do
+  resources :purchases
+  resources :groups
+  devise_for :users
```

Comment:
- [ ] It will be great to only configure the routes that are used by the application. For example, you are not using the edits anywhere in the application so it will be better to exclude these unused routes. You can learn more about this [here](https://edgeguides.rubyonrails.org/routing.html#shallow-nesting). You can use `: only` or  `: expect` to configure the routes